shared/modules/confirmation.utils.ts

```diff
@@ -0,0 +1,82 @@
+import { TransactionType } from '@metamask/transaction-controller';
+import { ApprovalType } from '@metamask/controller-utils';
+
+export const REDESIGN_SIGNATURE_APPROVAL_TYPES = [
+  ApprovalType.EthSignTypedData,
+  ApprovalType.PersonalSign,
+];
+
+const REDESIGN_USER_TRANSACTION_TYPES = [
+  TransactionType.contractInteraction,
+  TransactionType.deployContract,
+  TransactionType.tokenMethodApprove,
+  TransactionType.tokenMethodIncreaseAllowance,
+  TransactionType.tokenMethodSetApprovalForAll,
+  TransactionType.tokenMethodTransfer,
+  TransactionType.tokenMethodTransferFrom,
+  TransactionType.tokenMethodSafeTransferFrom,
+  TransactionType.simpleSend,
+];
+
+export const REDESIGN_DEV_TRANSACTION_TYPES = [
+  ...REDESIGN_USER_TRANSACTION_TYPES,
+];
+
+const isCorrectDeveloperTransactionType = (
+  transactionMetadataType: TransactionType,
+): boolean => REDESIGN_DEV_TRANSACTION_TYPES.includes(transactionMetadataType);
+
+const isCorrectUserTransactionType = (
+  transactionMetadataType: TransactionType,
+): boolean => REDESIGN_USER_TRANSACTION_TYPES.includes(transactionMetadataType);
+
+const isCorrectSignatureApprovalType = (approvalType: ApprovalType): boolean =>
+  REDESIGN_SIGNATURE_APPROVAL_TYPES.includes(approvalType);
+
+const shouldUseRedesignForTransactionsDeveloperMode = (
+  isRedesignedConfirmationsDeveloperEnabled: boolean,
+  transactionMetadataType: TransactionType,
+): boolean =>
+  (process.env.ENABLE_CONFIRMATION_REDESIGN === 'true' ||
+    isRedesignedConfirmationsDeveloperEnabled) &&
+  isCorrectDeveloperTransactionType(transactionMetadataType);
+
+const shouldUseRedesignForTransactionsUserMode = (
+  isRedesignedTransactionsUserSettingEnabled: boolean,
+  transactionMetadataType: TransactionType,
+): boolean =>
+  isRedesignedTransactionsUserSettingEnabled &&
+  isCorrectUserTransactionType(transactionMetadataType);
+
+export const shouldUseRedesignForTransactions = (
+  transactionMetadataType: TransactionType,
+  isRedesignedTransactionsUserSettingEnabled: boolean,
```

Comment:
Since these are exported utils using multiple boolean arguments, would it be more readable to use an options object argument?

Reply:
Updated it, but maybe we should think about adding the max-params rule -> https://eslint.org/docs/latest/rules/max-params.. otherwise it is up to the developer flavour 😅